p2p/switch.go

```diff
@@ -109,6 +119,16 @@ func NewSwitch(cfg *config.P2PConfig, options ...SwitchOption) *Switch {
 	return sw
 }
 
+// SwitchFilterTimeout sets the timeout used for peer filters.
+func SwitchFilterTimeout(timeout time.Duration) SwitchOption {
+	return func(sw *Switch) { sw.filterTimeout = timeout }
+}
+
+// SwitchPeerFilters sets the filters for rejection of new peers.
+func SwitchPeerFilters(filters ...PeerFilterFunc) SwitchOption {
```

Comment:
We should standardize, I'm strongly against the current form, as it is ambigious, from a callers perspective it looks like this `p2p.WithMetrics` which would make one believe this is on the package level. Ideally it, even if more verbose, options hould have the `SwitchOption` prefix, so that other components in the same package can expose their options without danger of collision. Would that make sense? /cc @melekes

Reply:
If there are multiple publicly exposed structs within a package, it makes sense to prefix options. SwitchPeerFilters looks OK to me